tests/integration/DOM-test.js

```diff
@@ -184,3 +185,19 @@ test('Clicking modal overlay', function(assert) {
   });
 
 });
+
+test('Pressing escape', function(assert) {
+
+  visit('/');
+
+  showModal(templateName);
+
+  andThen(function() {
+    keyEvent(inspect('overlay'), 'keydown', escapeKeyCode);
```

Comment:
keyEvent is a sync helper so you shouldn't need to use nested `andThen`s. I haven't used it before but try the following:

```
visit('/');

showModal(templateName);

keyEvent(inspect('overlay'), 'keydown', escapeKeyCode);

andThen(function() {
  assert.ok(/* ... */)
});    
```

Sorry if my first comment wasn't very clear.